lib/src/text_controller.dart

```diff
@@ -0,0 +1,78 @@
+part of 'hooks.dart';
+
+class _TextEditingControllerHookCreator {
+  const _TextEditingControllerHookCreator();
+
+  /// Creates a [TextEditingController] that will be disposed automatically.
+  ///
+  /// The [text] parameter can be used to set the initial value of the
+  /// controller.
+  TextEditingController call({String text, List<Object> keys}) {
+    return Hook.use(_TextEditingControllerHook(text, null, keys));
+  }
+
+  /// Creates a [TextEditingController] from the initial [value] that will
+  /// be disposed automatically.
+  TextEditingController fromValue(TextEditingValue value, [List<Object> keys]) {
+    return Hook.use(_TextEditingControllerHook(null, value, keys));
+  }
+}
+
+/// Functions to create a text editing controller, either via an initial
+/// text or an initial [TextEditingValue].
+///
+/// To use a [TextEditingController] with an optional initial text, use
+/// [_TextEditingControllerHookCreator.call]:
+/// ```dart
+/// final controller = useTextEditingController(text: 'initial text');
+/// ```
+///
+/// To use a [TextEditingController] with an optional inital value, use
+/// [_TextEditingControllerHookCreator.fromValue]:
+/// ```dart
+/// final controller = useTextEditingController
+///   .fromValue(TextEditingValue.empty);
+/// ```
```

Comment:
```suggestion
/// ```
///
/// See also:
/// - [TextEditingController], which this hooks creates.
```